src/IonEvent.ts

```diff
@@ -459,20 +514,21 @@ class IonListEvent extends AbsIonContainerEvent {
 
     }
 
-    valueEquals(expected: IonListEvent): boolean {
-        if (!(expected instanceof IonListEvent)) return false;
+    valueCompare(expected: IonListEvent): ComparisonResult {
+        if (!(expected instanceof IonListEvent)) return new ComparisonResult(ComparisonResultType.NOT_EQUAL, "Event types don't match");
         let container = this.ionValue;
         let expectedContainer = expected.ionValue;
-        if (container.length !== expectedContainer.length) return false;
+        if (container.length !== expectedContainer.length) return new ComparisonResult(ComparisonResultType.NOT_EQUAL, "List length don't match");
         for (let i: number = 0; i < container.length; i++) {
             let child = container[i];
-            if (!child.equals(expectedContainer[i])) {
-                return false;
+            if (child.compare(expectedContainer[i]).result == ComparisonResultType.NOT_EQUAL) {
+                return new ComparisonResult(ComparisonResultType.NOT_EQUAL, child.ionValue + " vs. " +
+                    expectedContainer[i].ionValue, i + 1, i + 1);
```

Comment:
Why +1 ?

Reply:
Considering the first event would be container start (`LIST`) and then I am rotating over its child events.